src/async_fuse/memfs/cache/memory_cache.rs

```diff
@@ -2,27 +2,373 @@
 
 use async_trait::async_trait;
 use clippy_utilities::OverflowArithmetic;
+use hashlink::LinkedHashSet;
 use lockfree_cuckoohash::{pin, LockFreeCuckooHash as HashMap};
-use std::collections::HashMap as StdHashMap;
-use std::sync::Arc;
-use tokio::sync::RwLock;
+use std::collections::{BTreeSet, HashMap as StdHashMap};
+use std::num::NonZeroUsize;
+use std::sync::{Arc, Weak};
+use std::time::Duration;
+use tokio::select;
+use tokio::sync::{mpsc, oneshot, RwLock};
+use tracing::warn;
 
 use super::policy::EvictPolicy;
 use super::{Block, BlockCoordinate, BlockId, Storage};
 use crate::async_fuse::fuse::protocol::INum;
 
-/// The file-level cache.
-type FileCache = Arc<RwLock<StdHashMap<BlockId, Block>>>;
-
 /// Merge the content from `src` to `dst`. This will set `dst` to be dirty.
 fn merge_two_blocks(src: &Block, dst: &mut Block) {
-    dst.set_dirty();
+    dst.set_dirty(true);
     dst.update(src);
 }
 
+/// A command sent from `MemoryCache` to the write back task.
```

Comment:
Split to another rust file with write_back_task related.

Reply:
It's not convenient to access private fields in `MemoryCache` from write back task, and make those fields `pub` is worse practice, therefore, I insist to hold them in the same file.